### ISSUE TITLE
Need to delete components and readd to reorder on page but wrong components get deleted.

### ISSUE THREAD

**Issue by delaynesr** [XGovFormBuilder/digital-form-builder#758]
**Describe the bug**
When I need to reorder components I have to delete the one in the wrong place to add it again after. I open the one I want to delete, click delete then it goes back to my page and the one I said to delete is still there and the other one is gone.

**To Reproduce**

1. Click on component on page I want to delete
2. Click delete at bottom of page
3. return to page, see error
4. Deleted component still there, unopened component beneath is gone

**Expected behavior**
I expected the one that I deleted to be the one to disappear. 

**Screenshots**

**Repository**

- designer

**Desktop (please complete the following information):**

- OS: windows 11
- Browser: microsoft edge
- Version 98.0.1108.50

**Additional context**

**Comment by JamesBoadi** [comment]
Which component failed to delete? Does this behaviour persist for all components?

**Comment by delaynesr** [comment]
Hi James,

I don't think it is related to the components. I just made a new page to
check and it happens where there are 3 components (or more maybe, I only
tested 3) and you try to delete the one in the middle. Then it
automatically deletes the one at the bottom (the most recent component
added) rather than the one in the middle (2nd component added).

Hope that makes sense.

Cheers,
Delayne

**Comment by JamesBoadi** [comment]
Yes I can see the issue now, currently components can only be deleted post order, not inorder. Probably a bug we will address in the future
